## Análise explanatória Empresas Ativas (SIMPLES)

#### Realização de uma análise explanatória detalhada dos dados das empresas ativas da cidade do Recife, verificando possíveis inconsistencias ou duplicações nos registros.

##### 🟩🟩 Importando Bibliotecas

In [2]:
import pandas as pd
import numpy as np


#### 🟩🟩 Chamando o arquivo

In [7]:
import os
# ID do documento do drive
FILE_ID = "1T5Dsh4MGd0aQv9Hp_VFu3ak8izovAej2"
OUTPUT_PATH = r'C:\Users\carli\Desktop\EmpresasAtivas\datasets\EmpresasAtivas.xlsx'

# Pasta para salvar os arquivos CSV
OUTPUT_FOLDER = r'C:\Users\carli\Desktop\EmpresasAtivas\datasets'

# Criando a pasta de saída, se não existir
os.makedirs(OUTPUT_FOLDER, exist_ok=True)

# URL do Google Drive com o ID 
GDRIVE_URL = f"https://drive.google.com/uc?id={FILE_ID}"

# Baixando o arquivo
try:
    gdown.download(GDRIVE_URL, OUTPUT_PATH, quiet=False)
    print(f"Arquivo salvo em: {OUTPUT_PATH}")
except Exception as e:
    print(f"Erro ao baixar o arquivo: {e}")


Downloading...
From (original): https://drive.google.com/uc?id=1T5Dsh4MGd0aQv9Hp_VFu3ak8izovAej2
From (redirected): https://drive.google.com/uc?id=1T5Dsh4MGd0aQv9Hp_VFu3ak8izovAej2&confirm=t&uuid=292348e6-44c5-4b2d-9b78-5de094ff8b36
To: C:\Users\carli\Desktop\EmpresasAtivas\datasets\EmpresasAtivas.xlsx
100%|██████████| 123M/123M [00:08<00:00, 14.9MB/s] 


Arquivo salvo em: C:\Users\carli\Desktop\EmpresasAtivas\datasets\EmpresasAtivas.xlsx


##### 🟩🟩 Dividindo o arquivo EmpresasAtivas.xlsx em múltiplos arquivos csv

In [11]:
import pandas as pd
import os

# Caminho do arquivo Excel com várias abas
INPUT_PATH = r'C:\Users\carli\Desktop\EmpresasAtivas\datasets\EmpresasAtivas.xlsx'

# Pasta para salvar os arquivos CSV
OUTPUT_FOLDER = r'C:\Users\carli\Desktop\EmpresasAtivas\datasets\CSV_Separados'

# Verificando se o arquivo Excel existe
if not os.path.exists(INPUT_PATH):
    print(f"Erro: O arquivo Excel '{INPUT_PATH}' não foi encontrado.")
    exit()

# Criando a pasta de saída, se não existir
os.makedirs(OUTPUT_FOLDER, exist_ok=True)

try:
    # Lendo todas as abas do arquivo Excel
    print("Carregando o arquivo Excel...")
    excel_data = pd.read_excel(INPUT_PATH, sheet_name=None)  # `sheet_name=None` carrega todas as abas
    print(f"Arquivo carregado com sucesso! Abas encontradas: {list(excel_data.keys())}")
except Exception as e:
    print(f"Erro ao carregar o arquivo Excel: {e}")
    exit()

# Salvando cada aba como um arquivo CSV separado
print("Iniciando a separação das abas em arquivos CSV...")
for sheet_name, data in excel_data.items():
    try:
        # Nome do arquivo CSV baseado no nome da aba
        output_file = os.path.join(OUTPUT_FOLDER, f"{sheet_name}.csv")
        
        # Salvando a aba como CSV
        data.to_csv(output_file, index=False, encoding='utf-8')
        print(f"Aba '{sheet_name}' salva como CSV: {output_file}")
    except Exception as e:
        print(f"Erro ao salvar a aba '{sheet_name}' como CSV: {e}")

print("Processo concluído! Todas as abas foram processadas.")



Carregando o arquivo Excel...
Arquivo carregado com sucesso! Abas encontradas: ['ESTABELECIMENTOS', 'EMPRESAS', 'SOCIOS', 'CNAES', 'SIMPLES']
Iniciando a separação das abas em arquivos CSV...
Aba 'ESTABELECIMENTOS' salva como CSV: C:\Users\carli\Desktop\EmpresasAtivas\datasets\CSV_Separados\ESTABELECIMENTOS.csv
Aba 'EMPRESAS' salva como CSV: C:\Users\carli\Desktop\EmpresasAtivas\datasets\CSV_Separados\EMPRESAS.csv
Aba 'SOCIOS' salva como CSV: C:\Users\carli\Desktop\EmpresasAtivas\datasets\CSV_Separados\SOCIOS.csv
Aba 'CNAES' salva como CSV: C:\Users\carli\Desktop\EmpresasAtivas\datasets\CSV_Separados\CNAES.csv
Aba 'SIMPLES' salva como CSV: C:\Users\carli\Desktop\EmpresasAtivas\datasets\CSV_Separados\SIMPLES.csv
Processo concluído! Todas as abas foram processadas.


##### 🟩🟩 Padronizando o encoding

In [8]:
dados = pd.read_csv('df_simples.csv', encoding= 'UTF-8', sep=';')

FileNotFoundError: [Errno 2] No such file or directory: 'df_simples.csv'

##### 🟩🟩 Observando o dataframe

In [22]:
dados.head(5)

CNPJ BASICO OPTANTE PELO SIMPLES?  DATA DE ADESAO AO SIMPLES  \
0            0                   NAO                   20070701   
1        19357                   NAO                   20070701   
2        28607                   NAO                   20070701   
3        59281                   SIM                   20220101   
4        79526                   NAO                   20080101   

   DATA DE EXCLUSAO DO SIMPLES OPTANTE PELO MEI?  DATA DE ADESAO AO MEI  \
0                     20070701               NAO               20090701   
1                     20101231               NAO                      0   
2                     20111231               NAO                      0   
3                            0               NAO                      0   
4                     20131231               NAO                      0   

   DATA DE EXCLUSAO DO MEI  
0                 20090701  
1                        0  
2                        0  
3                        0  
4                        0

##### 🟩🟩 Alterando o nome da coluna 'OPTANTE PELO SIMPLES?' , 'OPTANTE PELO MEI?' para 'OPTANTE PELO SIMPLES' , 'OPTANTE PELO MEI?'

In [24]:
# Renomear a coluna
dados = dados.rename(columns={'OPTANTE PELO SIMPLES?': 'OPTANTE PELO SIMPLES'})

# Verificar se a coluna foi renomeada corretamente
print(dados.columns)

Index(['CNPJ BASICO', 'OPTANTE PELO SIMPLES', 'DATA DE ADESAO AO SIMPLES',
       'DATA DE EXCLUSAO DO SIMPLES', 'OPTANTE PELO MEI?',
       'DATA DE ADESAO AO MEI', 'DATA DE EXCLUSAO DO MEI'],
      dtype='object')


In [26]:
# Renomear a coluna
dados = dados.rename(columns={'OPTANTE PELO MEI?': 'OPTANTE PELO MEI'})

# Verificar se a coluna foi renomeada corretamente
print(dados.columns)

Index(['CNPJ BASICO', 'OPTANTE PELO SIMPLES', 'DATA DE ADESAO AO SIMPLES',
       'DATA DE EXCLUSAO DO SIMPLES', 'OPTANTE PELO MEI',
       'DATA DE ADESAO AO MEI', 'DATA DE EXCLUSAO DO MEI'],
      dtype='object')


##### 🟡 Não existe CNPJ 0 , DATA de exclusão 0 e ADESAO E EXCLUSÃO DO MEI 0 então serão convertidos para campos vazios. Datas serão tratados mais abaixo

In [31]:
 colunas_valornan = ['CNPJ BASICO', 'OPTANTE PELO SIMPLES', 'OPTANTE PELO MEI']
dados[colunas_valornan] = dados[colunas_valornan].replace(0, np.nan)

# Verificar o resultado
print(dados)

        CNPJ BASICO OPTANTE PELO SIMPLES  DATA DE ADESAO AO SIMPLES  \
0               NaN                  NAO                   20070701   
1           19357.0                  NAO                   20070701   
2           28607.0                  NAO                   20070701   
3           59281.0                  SIM                   20220101   
4           79526.0                  NAO                   20080101   
...             ...                  ...                        ...   
348719   97553418.0                  SIM                   20240101   
348720   97553524.0                  SIM                   20110714   
348721   97553797.0                  NAO                   20110714   
348722   97553983.0                  NAO                   20110714   
348723   97554422.0                  NAO                   20110714   

        DATA DE EXCLUSAO DO SIMPLES OPTANTE PELO MEI  DATA DE ADESAO AO MEI  \
0                          20070701              NAO               2

##### 🟩🟩 Verificando a quantidade de nulos por coluna (mostrando apenas as colunas com nulos)

In [34]:
# Contar os valores nulos por coluna
nulos_por_coluna = dados.isna().sum()

# Filtrar as colunas com valores nulos (maior que 0)
colunas_com_nulos = nulos_por_coluna[nulos_por_coluna > 0]

# Exibir as colunas com valores nulos
print(colunas_com_nulos)

CNPJ BASICO    1
dtype: int64


##### 🟩🟩 Verificando dados duplicados 

In [37]:
duplicatas = dados.duplicated()

# Exibir as linhas duplicadas (se houver)
print(dados[duplicatas])

Empty DataFrame
Columns: [CNPJ BASICO, OPTANTE PELO SIMPLES, DATA DE ADESAO AO SIMPLES, DATA DE EXCLUSAO DO SIMPLES, OPTANTE PELO MEI, DATA DE ADESAO AO MEI, DATA DE EXCLUSAO DO MEI]
Index: []


##### Não há dados duplicados na tabela simples 

##### 🟩🟩 Verificando os tipos dos dados 

In [41]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 348724 entries, 0 to 348723
Data columns (total 7 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   CNPJ BASICO                  348723 non-null  float64
 1   OPTANTE PELO SIMPLES         348724 non-null  object 
 2   DATA DE ADESAO AO SIMPLES    348724 non-null  int64  
 3   DATA DE EXCLUSAO DO SIMPLES  348724 non-null  int64  
 4   OPTANTE PELO MEI             348724 non-null  object 
 5   DATA DE ADESAO AO MEI        348724 non-null  int64  
 6   DATA DE EXCLUSAO DO MEI      348724 non-null  int64  
dtypes: float64(1), int64(4), object(2)
memory usage: 18.6+ MB


##### 🟡 Em uma analise por data type é preciso ajustar  'CNPJ BASICO', para object pois eles não possuem a função de soma visto que se trata de códigos
##### 🟡 Em uma analise por data type é preciso ajustar 'DATA DE ADESAO AO SIMPLES' ; 'DATA DE EXCLUSAO DO SIMPLES' ; 'DATA DE ADESAO AO MEI', 'DATA DE EXCLUSAO DO MEI' para o formato de data

##### 🟩🟩 Alterando Dtype para object 

In [45]:
conversao = ['CNPJ BASICO']

# Converter as colunas específicas de 'int' para 'object'
dados[conversao] = dados[conversao].astype('object')

# Verificar os tipos de dados após a conversão
dados.dtypes

CNPJ BASICO                    object
OPTANTE PELO SIMPLES           object
DATA DE ADESAO AO SIMPLES       int64
DATA DE EXCLUSAO DO SIMPLES     int64
OPTANTE PELO MEI               object
DATA DE ADESAO AO MEI           int64
DATA DE EXCLUSAO DO MEI         int64
dtype: object

In [47]:
# Definir as colunas específicas para verificar
colunas_data = ['DATA DE ADESAO AO SIMPLES' , 'DATA DE EXCLUSAO DO SIMPLES' , 'DATA DE ADESAO AO MEI', 'DATA DE EXCLUSAO DO MEI']
dados[colunas_data]

DATA DE ADESAO AO SIMPLES  DATA DE EXCLUSAO DO SIMPLES  \
0                        20070701                     20070701   
1                        20070701                     20101231   
2                        20070701                     20111231   
3                        20220101                            0   
4                        20080101                     20131231   
...                           ...                          ...   
348719                   20240101                            0   
348720                   20110714                            0   
348721                   20110714                     20180201   
348722                   20110714                     20121231   
348723                   20110714                     20231231   

        DATA DE ADESAO AO MEI  DATA DE EXCLUSAO DO MEI  
0                    20090701                 20090701  
1                           0                        0  
2                           0                        0  
3                           0                        0  
4                           0                        0  
...                       ...                      ...  
348719                      0                        0  
348720               20110714                        0  
348721               20110714                 20180201  
348722               20110714                 20110930  
348723               20110714                 20231231  

[348724 rows x 4 columns]

In [49]:
# Substituir valores '0' ou inválidos por NaT (Not a Time)
dados['DATA DE ADESAO AO SIMPLES'] = dados['DATA DE ADESAO AO SIMPLES'].replace('0', pd.NaT)
dados['DATA DE EXCLUSAO DO SIMPLES'] = dados['DATA DE EXCLUSAO DO SIMPLES'].replace('0', pd.NaT)
dados['DATA DE ADESAO AO MEI'] = dados['DATA DE ADESAO AO MEI'].replace('0', pd.NaT)
dados['DATA DE EXCLUSAO DO MEI'] = dados['DATA DE EXCLUSAO DO MEI'].replace('0', pd.NaT)

# Converter para o formato datetime, forçando o tratamento de valores inválidos
dados['DATA DE ADESAO AO SIMPLES'] = pd.to_datetime(dados['DATA DE ADESAO AO SIMPLES'], errors='coerce', format='%Y%m%d')
dados['DATA DE EXCLUSAO DO SIMPLES'] = pd.to_datetime(dados['DATA DE EXCLUSAO DO SIMPLES'], errors='coerce', format='%Y%m%d')
dados['DATA DE ADESAO AO MEI'] = pd.to_datetime(dados['DATA DE ADESAO AO MEI'], errors='coerce', format='%Y%m%d')
dados['DATA DE EXCLUSAO DO MEI'] = pd.to_datetime(dados['DATA DE EXCLUSAO DO MEI'], errors='coerce', format='%Y%m%d')

# Formatar as datas para DD/MM/AAAA
dados['DATA DE ADESAO AO SIMPLES'] = dados['DATA DE ADESAO AO SIMPLES'].dt.strftime('%d/%m/%Y')
dados['DATA DE EXCLUSAO DO SIMPLES'] = dados['DATA DE EXCLUSAO DO SIMPLES'].dt.strftime('%d/%m/%Y')
dados['DATA DE ADESAO AO MEI'] = dados['DATA DE ADESAO AO MEI'].dt.strftime('%d/%m/%Y')
dados['DATA DE EXCLUSAO DO MEI'] = dados['DATA DE EXCLUSAO DO MEI'].dt.strftime('%d/%m/%Y')

# Exibir as colunas para verificar a conversão
dados[['DATA DE ADESAO AO SIMPLES', 'DATA DE EXCLUSAO DO SIMPLES', 'DATA DE ADESAO AO MEI','DATA DE EXCLUSAO DO MEI']]


DATA DE ADESAO AO SIMPLES DATA DE EXCLUSAO DO SIMPLES  \
0                     01/07/2007                  01/07/2007   
1                     01/07/2007                  31/12/2010   
2                     01/07/2007                  31/12/2011   
3                     01/01/2022                         NaN   
4                     01/01/2008                  31/12/2013   
...                          ...                         ...   
348719                01/01/2024                         NaN   
348720                14/07/2011                         NaN   
348721                14/07/2011                  01/02/2018   
348722                14/07/2011                  31/12/2012   
348723                14/07/2011                  31/12/2023   

       DATA DE ADESAO AO MEI DATA DE EXCLUSAO DO MEI  
0                 01/07/2009              01/07/2009  
1                        NaN                     NaN  
2                        NaN                     NaN  
3                        NaN                     NaN  
4                        NaN                     NaN  
...                      ...                     ...  
348719                   NaN                     NaN  
348720            14/07/2011                     NaN  
348721            14/07/2011              01/02/2018  
348722            14/07/2011              30/09/2011  
348723            14/07/2011              31/12/2023  

[348724 rows x 4 columns]

##### 🟩🟩 Verificando datas de abertura e fechamento do simples e do MEI

In [52]:
# Filtrar linhas onde 'DATA DE ADESAO AO SIMPLES' é igual a 'DATA DE EXCLUSAO DO SIMPLES'
filtro_simples = dados['DATA DE ADESAO AO SIMPLES'] == dados['DATA DE EXCLUSAO DO SIMPLES']
resultado_simples = dados[filtro_simples]

# Exibir os resultados
print("Linhas onde Adesão e Exclusão do Simples são iguais:")


resultado_simples

Linhas onde Adesão e Exclusão do Simples são iguais:


CNPJ BASICO OPTANTE PELO SIMPLES DATA DE ADESAO AO SIMPLES  \
0              NaN                  NAO                01/07/2007   
109       191211.0                  NAO                01/07/2007   
134       209683.0                  NAO                01/01/2009   
191       260594.0                  NAO                01/07/2007   
299       360305.0                  NAO                01/07/2007   
...            ...                  ...                       ...   
348123  70090733.0                  NAO                01/01/2014   
348201  70179817.0                  NAO                01/01/2016   
348307  70221064.0                  NAO                01/07/2007   
348524  97531596.0                  NAO                11/07/2011   
348573  97535869.0                  NAO                01/01/2016   

       DATA DE EXCLUSAO DO SIMPLES OPTANTE PELO MEI DATA DE ADESAO AO MEI  \
0                       01/07/2007              NAO            01/07/2009   
109                     01/07/2007              NAO                   NaN   
134                     01/01/2009              NAO                   NaN   
191                     01/07/2007              NAO                   NaN   
299                     01/07/2007              NAO                   NaN   
...                            ...              ...                   ...   
348123                  01/01/2014              NAO                   NaN   
348201                  01/01/2016              NAO                   NaN   
348307                  01/07/2007              NAO                   NaN   
348524                  11/07/2011              NAO            11/07/2011   
348573                  01/01/2016              NAO                   NaN   

       DATA DE EXCLUSAO DO MEI  
0                   01/07/2009  
109                        NaN  
134                        NaN  
191                        NaN  
299                        NaN  
...                        ...  
348123                     NaN  
348201                     NaN  
348307                     NaN  
348524              11/07/2011  
348573                     NaN  

[1805 rows x 7 columns]

In [54]:
# Filtrar linhas onde 'DATA DE ADESAO AO MEI' é igual a 'DATA DE EXCLUSAO DO MEI'
filtro_mei = dados['DATA DE ADESAO AO MEI'] == dados['DATA DE EXCLUSAO DO MEI']
resultado_mei = dados[filtro_mei]



print("\nLinhas onde Adesão e Exclusão do MEI são iguais:")
resultado_mei


Linhas onde Adesão e Exclusão do MEI são iguais:


CNPJ BASICO OPTANTE PELO SIMPLES DATA DE ADESAO AO SIMPLES  \
0              NaN                  NAO                01/07/2007   
1170     1109916.0                  SIM                01/01/2013   
6468     4297984.0                  NAO                01/01/2008   
13702    7368636.0                  NAO                01/01/2021   
20388    9020772.0                  SIM                12/09/2007   
...            ...                  ...                       ...   
347029  57917761.0                  NAO                31/10/2024   
347463  57998770.0                  NAO                06/11/2024   
347472  57999901.0                  NAO                06/11/2024   
347858  69937951.0                  NAO                01/01/2009   
348524  97531596.0                  NAO                11/07/2011   

       DATA DE EXCLUSAO DO SIMPLES OPTANTE PELO MEI DATA DE ADESAO AO MEI  \
0                       01/07/2007              NAO            01/07/2009   
1170                           NaN              NAO            01/01/2015   
6468                    31/12/2017              NAO            01/01/2015   
13702                   06/04/2022              NAO            01/01/2019   
20388                          NaN              NAO            01/01/2015   
...                            ...              ...                   ...   
347029                  31/10/2024              NAO            31/10/2024   
347463                  06/11/2024              NAO            06/11/2024   
347472                  06/11/2024              NAO            06/11/2024   
347858                  31/12/2010              NAO            01/01/2011   
348524                  11/07/2011              NAO            11/07/2011   

       DATA DE EXCLUSAO DO MEI  
0                   01/07/2009  
1170                01/01/2015  
6468                01/01/2015  
13702               01/01/2019  
20388               01/01/2015  
...                        ...  
347029              31/10/2024  
347463              06/11/2024  
347472              06/11/2024  
347858              01/01/2011  
348524              11/07/2011  

[1928 rows x 7 columns]

##### 🟡 Existe 1805 linhas em que a data de adesão e exclusão do simples é a mesma
##### 🟡 Existe 1928 linhas em que a data de expedição e exclusão do MEI é a mesma. 

##### 🟩🟩 Salvando o dataframe estabelecimento 

In [ ]:
dados.to_parquet('Scripts_EmpresasAtivas/simples.parquet')

In [58]:
# Salva o DataFrame em formato CSV com separador e controle de casas decimais
dados.to_csv('Scripts_EmpresasAtivas/simples.csv', sep=';', decimal=',', encoding='UTF-8')